crates/query-engine/translation/src/translation/query/relationships.rs

```diff
@@ -35,87 +35,18 @@ pub fn translate_joins(
                 relationship_arguments: relationship.arguments.clone(),
             })?;
 
-            // create a from clause and get a reference of inner query.
-            let (target_collection, from_clause) = root::make_from_clause_and_reference(
-                &relationship.target_collection,
-                &arguments,
-                env,
-                state,
-                None,
-            )?;
-
             // process inner query and get the SELECTs for the 'rows' and 'aggregates' fields.
-            let select_set = super::translate_query(
+            let select_set = root::translate_query(
                 env,
                 state,
-                &target_collection,
-                &from_clause,
+                root::MakeFrom::Collection {
+                    name: relationship.target_collection.clone(),
+                    arguments: arguments.clone(),
+                },
```

Comment:
Instead of building a from clause and passing it, we build a structure describing how to build the from clause.